src/jwk.rs

```diff
@@ -271,6 +272,46 @@ impl AlgorithmParameters {
             _ => Err(unexpected_key_type_error!(KeyType::Octet, self.key_type())),
         }
     }
+
+    /// RFC 7638
+    pub fn thumbprint(
+        &self,
+        algo: &'static ring::digest::Algorithm,
+    ) -> Result<String, serde_json::error::Error> {
+        use serde::ser::SerializeMap;
+
+        use crate::serde_custom::{base64_url_uint, byte_sequence};
+
+        let mut serializer = serde_json::Serializer::new(Vec::new());
```

Comment:
Could you explain to me why you opted to "manually" serialize this instead of creating an enum and deriving the serialize methods for them?

Reply:
The serialization has to be byte-precise, and doing it manually shows precisely how the RFC is followed.
I see that biscuit enables ordered json through crate features, but that changes the behavior of the whole crate graph, which might hide problems instead of fixing them, so I preferred not to rely on that.

Reply:
Got it. Thanks!